libs/failsafe-controller/src/main/java/net/jodah/failsafe/AsyncFailsafeConfigDelegater.java

```diff
@@ -27,13 +32,14 @@
 
   public AsyncFailsafeConfigDelegater(FailsafeConfig<R, ?> master, Scheduler scheduler) {
     super(master, scheduler);
+    this.master = master;
     // copy all config from master
     super.retryPolicy = master.retryPolicy;
     super.circuitBreaker = master.circuitBreaker;
     super.fallback = master.fallback;
     super.listeners = master.listeners;
-    super.listenerRegistry = master.listenerRegistry;
-    this.master = master;
+    // make sure we use the master registry while ensuring it was initialized
+    super.listenerRegistry = master.registry();
```

Comment:
ℹ️ By doing a call to `master.registry()`, I no longer need to override any of the listener adding methods or, as in the previous version, the `registry()` method. The only listener adding methods overridden here are the asynchronous listeners such that we can later monitor their execution.